src/components/table-hoc/TableWithSort.tsx

```diff
@@ -8,7 +8,8 @@ import {ITableWithSortState} from './reducers/TableWithSortReducers';
 import {ITableHOCOwnProps} from './TableHOC';
 
 export interface ITableWithSortConfig {
-    sort: (sortKey: string, isAsc: boolean, a: any, b: any) => number;
+    sort?: (sortKey: string, isAsc: boolean, a: any, b: any) => number;

```

Comment:
Can it be generic? 

`sort<T = any>(sortKey: string, isAsc: boolean, a: T, b: T)`

Reply:
Same thing as the other method for filtering, since it's any you can use your own type (see examples)